### ISSUE TITLE
UI freezes for about 1 second when user clicks "Add to contacts" in the contact dialogue

### ISSUE THREAD

**Issue by John-44** [status-im/status-desktop#3529]
when a User clicks "Add to contacts" in the contact dialogue, the UI freezes for about 1 second before the dialogue is dismissed.  It freezes for just long enough for the user to think something is broken and try clicking again.

Desired resolution: when a user clicks "Add to contacts" the contact dialogue should close instantly without delay

**Comment by michaelsbradleyjr** [comment]
I've spent some time analyzing this one, thinking about if/how `task_runner` or some other approach could be used in the existing codebase to get the work in `addContact` off the main thread:

https://github.com/status-im/status-desktop/blob/master/src/app/profile/views/contacts.nim#L200

https://github.com/status-im/status-lib/blob/master/status/contacts.nim#L94-L123

The difficulty is that, per the code in that second link, adding a contact involves reads and writes to the data model that lives on the main thread, interleaved with calls to status-go and emission of EventEmitter events.

So it's all interwoven synchronous calls. Data model access (read or write) *must* be done on the thread hosting the model because of constraints of Nim v1.2's memory model. The same is true for usage of EventEmitter. Finally, since the work is done in the context of status-lib, we don't have a ready way to signal back from `task_runner` if we attempted to move the status-go calls to a `task_runner` worker pool, i.e. in status-desktop we rely on Qt slot/signal to pass task results back to the main thread but that's not available in the context of status-lib.

Hopefully, the changes that will be made re: #3651 will improve the performance and reduce or eliminate the UI freeze when adding a contact.
